File: MasonryTableViewCell.m
//
//  MasonryTableViewCell.m
//  LuoMiPhone
//
//  Created by Tim Geng on 4/18/15.
//  Copyright (c) 2015 GF. All rights reserved.
//

#import "MasonryTableViewCell.h"
#import "MasonryCollectionViewCell.h"
#import "MKMasonryViewLayout.h"

@interface MasonryTableViewCell ()<UICollectionViewDataSource,UICollectionViewDelegate,MKMasonryViewLayoutDelegate>

@end

@implementation MasonryTableViewCell

- (void)awakeFromNib {
    // Initialization code
    
    [self.collectView registerNib:[UINib nibWithNibName:NSStringFromClass([MasonryCollectionViewCell class]) bundle:nil] forCellWithReuseIdentifier:@"MasonryCollectionViewCellIdentifier"];
    self.collectView.delegate = self;
    self.collectView.dataSource = self;

}

- (CGFloat) collectionView:(UICollectionView*) collectionView
                    layout:(MKMasonryViewLayout*) layout
  heightForItemAtIndexPath:(NSIndexPath*) indexPath{
    if (indexPath.row == 0 || indexPath.row == 1) {
        return 130 * 2 + 3;
    }
    return 130;
}

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section{
    return 7;
}

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView{
    return 1;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath{
    MasonryCollectionViewCell *cell = (MasonryCollectionViewCell *)[collectionView dequeueReusableCellWithReuseIdentifier:@"MasonryCollectionViewCellIdentifier" forIndexPath:indexPath];
    if (indexPath.row ==0 ) {
        cell.imageView.image = [UIImage imageNamed:@"MasonryPic1"];
    }
    if (indexPath.row ==1) {
        cell.imageView.image = [UIImage imageNamed:@"MasonryPic2"];
    }
    //cell.title.text = [NSString stringWithFormat:@"%ld",(long)indexPath.row ];
    return cell;
}


@end

File: MasonryTableViewCell.swift
//
//  MasonryTableViewCell.swift
//  LuoMiPhone
//
//  Created by Tim Geng on 4/9/15.
//  Copyright (c) 2015 GF. All rights reserved.
//

import UIKit

@objc class MasonryTableViewCell: UITableViewCell,UICollectionViewDataSource,UICollectionViewDelegate {
    
@IBOutlet var collectionView: UICollectionView!;
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.collectionView .registerNib(UINib(nibName: "MasonryCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "MasonryCollectionViewCellIdentifier");
        // Initialization code
    }

    override func setSelected(selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }
    
    func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int{
        return 7;
    }
    
    func numberOfSectionsInCollectionView(collectionView: UICollectionView?) -> Int {
        return 1
    }
    
    func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell{
        let cell : MasonryCollectionViewCell = collectionView.dequeueReusableCellWithReuseIdentifier("MasonryCollectionViewCellIdentifier", forIndexPath: indexPath) as MasonryCollectionViewCell
        return cell;
    }
    
}
